app/controllers/posts_controller.rb

```diff
@@ -1,36 +1,45 @@
 # frozen_string_literal: true
 
 class PostsController < ApplicationController
+  before_action :load_post!, only: %i[show update destroy]
+  before_action :load_posts!, only: :index
   def index
-    @posts = Post.includes(:categories, :user, :organization).where(organization_id: params[:organization_id])
-
     if params[:category_ids].present?
       category_ids = params[:category_ids].map(&:to_i)
       @posts = @posts.joins(:categories).where(categories: { id: category_ids }).distinct
     end
   end
 
   def create
-    post = Post.new(post_params)
-    post.categories = Category.where(id: params[:post][:category_ids]) if params[:post][:category_ids].present?
-
-    post.save!
+    Post.create(post_params)
```

Comment:
```suggestion
    Post.create!(post_params)
```

Please read the chapter in LRRB on why we use bang operator.

Reply:
Yes, I have read. Bang operator is used to raise exceptions.

Reply:
You can merge after the above fix.